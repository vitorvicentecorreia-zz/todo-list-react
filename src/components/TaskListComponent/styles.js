import styled from 'styled-components';

export const TaskList = styled.ul`
    width: 100%;
    height: 70%; 
    overflow-y: auto;
`
export const ListItem = styled.li`
    margin: 10px auto;
    width: 80%;
    height: 40px;
    display: flex;
    justify-content: space-between;
    align-items: center;
`
export const ItemText = styled.div`
    font-size: 16px;
    font-weight: bold;
    width: 90%;
    display: flex;
    align-items: center;

    ${props => props.completed && `
        color: #d4d4d4;
        text-decoration: line-through;
        text-decoration-thickness: 10px;
    `} 
`
export const HiddenCheckbox = styled.input.attrs({type: 'checkbox'})`
    border: 0;
    clip: rect(0 0 0 0);
    clippath: inset(50%);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    width: 1px;
`
export const StyledCheckbox = styled.div`
    display: inline-block;
    width: 16px;
    height: 16px;
    background: transparent;
    border-radius: 3px;
    transition: all 150ms;
    border: 1px solid #cccccc;

`
export const CheckboxContainer = styled.div`
    display: inline-block;
    vertical-align: middle;
    width: 10%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
`

export const Icon = styled.svg`
  fill: none;
  stroke: #b617ea;
  stroke-width: ${props => props.checked ? '2px' : '0px'};
`

export const Trash = styled.div`
    width: 10%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
`
