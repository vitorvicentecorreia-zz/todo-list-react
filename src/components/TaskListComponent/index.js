import React from 'react';
import { FaRegTrashAlt } from "react-icons/fa";

import { TaskList, ListItem, ItemText, CheckboxContainer, HiddenCheckbox, StyledCheckbox, Icon, Trash } from './styles';


export default function TaskListComponent(props) {

	return (
        <TaskList>
            {props.tasks && props.tasks.map((task) => (
                <ListItem key={task.id}>
                    <ItemText completed={true}>{task.text}</ItemText>
                    <CheckboxContainer>
                        <HiddenCheckbox />
                        <StyledCheckbox>
                            <Icon checked={true} viewBox="0 0 24 24">
                                <polyline points="20 6 9 17 4 12" />
                            </Icon>
                        </StyledCheckbox>
                    </CheckboxContainer>
                    <Trash>
                        <FaRegTrashAlt color="#b617ea" /> 
                    </Trash>
                </ListItem>
            ))}
        </TaskList>
	);
}
