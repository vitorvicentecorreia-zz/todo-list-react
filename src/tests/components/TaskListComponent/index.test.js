import React from 'react'
import { shallow } from 'enzyme'

import TaskListComponent from '../../../components/TaskListComponent'

it('Testar renderização do componente de lista de tarefas', () => {
    const tasks = [
        { id: 1, text: 'task 1'},
        { id: 2, text: 'task 2'}
    ]
    const component = shallow(<TaskListComponent tasks={tasks} />)
    expect(component).toMatchSnapshot();
})