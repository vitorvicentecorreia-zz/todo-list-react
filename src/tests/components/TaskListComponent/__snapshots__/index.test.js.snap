// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparar snapshots do componente TaskListComponent 1`] = `
<styled.ul>
  <styled.li
    key="1"
  >
    task 1
  </styled.li>
  <styled.li
    key="2"
  >
    task 2
  </styled.li>
</styled.ul>
`;

exports[`Testar renderização do componente de lista de tarefas 1`] = `
<styled.ul>
  <styled.li
    key="1"
  >
    task 1
  </styled.li>
  <styled.li
    key="2"
  >
    task 2
  </styled.li>
</styled.ul>
`;
